lsp/test/text_document_tests.ml

```diff
@@ -103,12 +103,12 @@ let%expect_test "replace first line" =
 
 let%expect_test "beyond max char" =
   let range = tuple_range (0, 0) (0, 100) in
-  test "foo\nbar\n" range ~change:"baz\n";
+  test "foo\nbar\n" range ~change:"baz";
   [%expect {|
     UTF16:
-    baz\nbar\n
+    bazbar\n
```

Comment:
According to my understanding of the [spec](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#range), the edited text should be `baz\nbar\n`, meaning that a range spanning beyond max char should not include the line ending `\n` (?)

Reply:
ah yes, my bad. pushing a fix.